src/rust/cryptography-x509/src/common.rs

```diff
@@ -17,6 +17,24 @@ impl AlgorithmIdentifier<'_> {
     pub fn oid(&self) -> &asn1::ObjectIdentifier {
         self.params.item()
     }
+
+    pub fn is_rsa(&self) -> bool {
+        matches!(
+            self.params,
+            AlgorithmParameters::Rsa(_)
+                | AlgorithmParameters::RsaWithSha1(_)
+                | AlgorithmParameters::RsaWithSha1Alt(_)
+                | AlgorithmParameters::RsaWithSha224(_)
+                | AlgorithmParameters::RsaWithSha256(_)
+                | AlgorithmParameters::RsaWithSha384(_)
+                | AlgorithmParameters::RsaWithSha512(_)
+                | AlgorithmParameters::RsaWithSha3_224(_)
+                | AlgorithmParameters::RsaWithSha3_256(_)
+                | AlgorithmParameters::RsaWithSha3_384(_)
+                | AlgorithmParameters::RsaWithSha3_512(_)
+                | AlgorithmParameters::RsaPss(_)
+        )
+    }
```

Comment:
This combines key algorithms and signature algorithms, which I think is pretty confusing

Reply:
Nixed the signature algorithms and renamed to `is_rsa_key` (but maybe it should be `is_rsa_key_algo`?)